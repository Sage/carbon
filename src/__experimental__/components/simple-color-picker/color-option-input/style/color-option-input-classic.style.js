import { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';
import StyledColorSampleBox from '../../color-sample-box/style/color-sample-box.style';

export default ({ theme }) => theme.name === THEMES.classic && css`
  &:checked + ${StyledColorSampleBox}, 
  &:focus + ${StyledColorSampleBox} {
    box-shadow: none;
    border: 1px solid #003349;
  }
`;
