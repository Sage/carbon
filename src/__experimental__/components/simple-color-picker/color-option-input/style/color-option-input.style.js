import styled from 'styled-components';
import baseTheme from '../../../../../style/themes/base';
import { Input as ColorOptionInput } from '../../../input';
import StyledColorSampleBox from '../../color-sample-box/style/color-sample-box.style';
import ColorOptionInputClassicStyle from './color-option-input-classic.style';

const StyledColorOptionInput = styled(ColorOptionInput)`
  position: absolute;
  opacity: 0;
  height: 56px;
  width: 56px;
  margin: 0;

  &:hover {
    cursor: pointer;
  }

  &:focus + ${StyledColorSampleBox} {
    box-shadow: inset 0px 0px 0px 3px ${({ theme }) => theme.colors.white};
    border: 2px solid ${({ theme }) => theme.colors.focus};
  }

  ${ColorOptionInputClassicStyle}
`;

StyledColorOptionInput.defaultProps = {
  theme: baseTheme
};

export default StyledColorOptionInput;
