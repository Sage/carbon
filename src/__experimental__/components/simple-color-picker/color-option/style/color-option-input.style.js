import styled from 'styled-components';
import baseTheme from '../../../../../style/themes/base';
// eslint-disable-next-line import/no-named-as-default
import Input from '../../../input';
import ColorOptionInputClassicStyle from './color-option-input-classic.style';
import StyledColorSampleBox from './color-sample-box.style';

const StyledColorOptionInput = styled(Input)`
  position: absolute;
  opacity: 0;
  height: 56px;
  width: 56px;
  margin: 0;

  &:hover {
    cursor: pointer;
  }

  &:checked + ${StyledColorSampleBox} {
    box-shadow: inset 0px 0px 0px 3px ${({ theme }) => theme.colors.white};
    border: 2px solid ${({ theme }) => theme.colors.primary};
  }

  &:focus + ${StyledColorSampleBox} {
    box-shadow: inset 0px 0px 0px 3px ${({ theme }) => theme.colors.white};
    border: 2px solid ${({ theme }) => theme.colors.focus};
  }

  ${ColorOptionInputClassicStyle}
`;

StyledColorOptionInput.defaultProps = {
  theme: baseTheme
};

export default StyledColorOptionInput;
