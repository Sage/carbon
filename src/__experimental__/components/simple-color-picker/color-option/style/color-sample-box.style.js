import styled from 'styled-components';
import PropTypes from 'prop-types';
import colorSampleBoxClassicStyle from './color-sample-box-classic.style';

const StyledColorSampleBox = styled.div`
  height: 100%;
  width: 100%;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 2px solid transparent;
  background-color: ${({ color }) => color};

  ${colorSampleBoxClassicStyle}
`;

StyledColorSampleBox.propTypes = {
  color: PropTypes.string
};

export default StyledColorSampleBox;
