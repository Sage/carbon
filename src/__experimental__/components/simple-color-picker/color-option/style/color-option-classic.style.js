import { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';

export default ({ theme }) => theme.name === THEMES.classic
  && css`
    margin-right: 1px;
    margin-bottom: 1px;
  `;
