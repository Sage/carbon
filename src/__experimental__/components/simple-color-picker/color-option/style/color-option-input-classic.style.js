import { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';
import StyledColorSampleBox from './color-sample-box.style';
import StyledTickIcon from './tick-icon.style';

export default ({ theme }) => theme.name === THEMES.classic
  && css`
    &:checked + ${StyledColorSampleBox} {
      box-shadow: none;
      border: 1px solid #003349;
      ${StyledTickIcon} {
        display: block;
      }
    }

    &:focus + ${StyledColorSampleBox} {
      border: 1px solid #003349;
      box-shadow: none;
    }
  `;
