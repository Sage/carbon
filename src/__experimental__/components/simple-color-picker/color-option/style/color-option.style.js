import styled, { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';

const StyledColorOption = styled.li`
  width: 56px;
  height: 56px;
  margin-right: 2px;
  margin-bottom: 2px;
  list-style: none;

  &:hover {
    cursor: pointer;
  }

  ${({ theme }) => theme.name === THEMES.classic && css`
    margin-right: 1px;
    margin-bottom: 1px;
  `};
`;

export default StyledColorOption;
