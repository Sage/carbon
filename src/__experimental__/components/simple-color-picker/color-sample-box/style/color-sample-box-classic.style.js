import { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';

export default ({ theme }) => theme.name === THEMES.classic && css`
  border: 1px solid transparent;
  ${({ color }) => (color === 'transparent' || color === 'none') && css`
    border-color: #b3c2c8;
  `}
`;
