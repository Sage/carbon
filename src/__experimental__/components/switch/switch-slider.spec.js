import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import { css, ThemeProvider } from 'styled-components';
import { assertStyleMatch, carbonThemesJestTable } from '../../../__spec_helper__/test-utils';
import Loader from '../../../components/loader/loader.component';
import SwitchSlider from './switch-slider.component';
import SwitchSliderPanel from './switch-slider-panel.style';
import { baseTheme } from '../../../style/themes';

describe('SwitchSlider', () => {
  describe('base theme', () => {
    it('renders as expected', () => {
      expect(render()).toMatchSnapshot();
    });

    describe('Panel content', () => {
      describe('default', () => {
        const panels = render().root.findAllByType(SwitchSliderPanel);

        it('renders the text "OFF" in the panel', () => {
          expect(panels[0].props.children).toBe('OFF');
        });

        it('renders only one panel', () => {
          expect(panels.length).toBe(1);
        });
      });

      describe('when checked=true', () => {
        const panels = render({ checked: true }).root.findAllByType(SwitchSliderPanel);

        it('renders the text "ON" in the panel', () => {
          expect(panels[0].props.children).toBe('ON');
        });

        it('renders only one panel', () => {
          expect(panels.length).toBe(1);
        });
      });

      describe('when loading=true', () => {
        const panels = render({ loading: true }).root.findAllByType(SwitchSliderPanel);

        it('renders a Loader in the first panel', () => {
          expect(panels[0].props.children.type).toBe(Loader);
        });

        it('renders only one panel', () => {
          expect(panels.length).toBe(1);
        });
      });
    });

    describe('when checked=true', () => {
      const wrapper = render({ checked: true }).toJSON();

      it('applies the correct base styles', () => {
        assertStyleMatch({
          backgroundColor: baseTheme.colors.primary
        }, wrapper);
      });

      it('applies the correct ::before styles', () => {
        assertStyleMatch({
          transform: 'translateX(36px)'
        }, wrapper, { modifier: '::before' });
      });
    });

    describe('when disabled=true', () => {
      const wrapper = render({ disabled: true }).toJSON();

      it('applies the correct base styles', () => {
        assertStyleMatch({
          backgroundColor: baseTheme.disabled.background
        }, wrapper);
      });

      it('applies the correct ::before styles', () => {
        assertStyleMatch({
          opacity: '0.8'
        }, wrapper, { modifier: '::before' });
      });

      it('applies the correct SwitchSliderPanel styles', () => {
        assertStyleMatch({
          color: baseTheme.disabled.disabled
        }, wrapper, { modifier: css`${SwitchSliderPanel}` });
      });
    });

    describe('when checked=true && disabled=true', () => {
      const wrapper = render({ checked: true, disabled: true }).toJSON();

      it('applies the correct SwitchSliderPanel styles', () => {
        assertStyleMatch({
          color: baseTheme.colors.white
        }, wrapper, { modifier: css`${SwitchSliderPanel}` });
      });
    });

    describe('when size=large', () => {
      describe('default', () => {
        const wrapper = render({ size: 'large' }).toJSON();

        it('applies the correct ::before styles', () => {
          assertStyleMatch({
            height: '36px',
            width: '36px'
          }, wrapper, { modifier: '::before' });
        });
      });

      describe('and checked=true', () => {
        const wrapper = render({ checked: true, size: 'large' }).toJSON();

        it('applies the correct ::before styles', () => {
          assertStyleMatch({
            transform: 'translateX(38px)'
          }, wrapper, { modifier: '::before' });
        });
      });
    });
  });

  describe.each(carbonThemesJestTable)('when the theme is set to %s', (themeName, theme) => {
    describe('default', () => {
      const wrapper = renderWithTheme({}, theme).toJSON();

      it('applies the correct base styles', () => {
        assertStyleMatch({
          backgroundColor: theme.switch.off
        }, wrapper);
      });

      it('applies the correct ::before styles', () => {
        assertStyleMatch({
          backgroundColor: theme.colors.white
        }, wrapper, { modifier: '::before' });
      });
    });

    describe('and checked=true', () => {
      const wrapper = renderWithTheme({ checked: true }, theme).toJSON();

      it('applies the correct base styles', () => {
        assertStyleMatch({
          backgroundColor: theme.colors.primary
        }, wrapper);
      });
    });

    describe('and disabled=true', () => {
      const wrapper = renderWithTheme({ disabled: true }, theme).toJSON();

      it('applies the correct base styles', () => {
        assertStyleMatch({
          backgroundColor: theme.disabled.background
        }, wrapper);
      });

      it('applies the correct SwitchSliderPanel styles', () => {
        assertStyleMatch({
          color: theme.disabled.disabled
        }, wrapper, { modifier: css`${SwitchSliderPanel}` });
      });
    });

    describe('when checked=true && disabled=true', () => {
      const wrapper = renderWithTheme({ checked: true, disabled: true }, theme).toJSON();

      it('applies the correct base styles', () => {
        assertStyleMatch({
          backgroundColor: theme.colors.disabled
        }, wrapper);
      });

      it('applies the correct SwitchSliderPanel styles', () => {
        assertStyleMatch({
          color: theme.colors.white
        }, wrapper, { modifier: css`${SwitchSliderPanel}` });
      });
    });
  });
});

function render(props) {
  return TestRenderer.create(<SwitchSlider { ...props } />);
}

function renderWithTheme(props, theme, renderer = TestRenderer.create) {
  return renderer(
    <ThemeProvider theme={ theme }>
      <SwitchSlider { ...props } />
    </ThemeProvider>
  );
}
