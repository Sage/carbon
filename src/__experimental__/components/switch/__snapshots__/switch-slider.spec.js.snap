// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSlider base theme renders as expected 1`] = `
.c2 {
  border: 0;
  color: #FFFFFF;
  margin: auto;
  margin-top: 5px;
}

.c2[type='on'] {
  margin-left: 9px;
}

.c2[type='off'] {
  color: rgba(0,0,0,0.9);
  margin-right: 6px;
}

.c0 {
  background-color: #CCD6DB;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  font-weight: bold;
  height: 24px;
  left: 0;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  position: absolute;
  top: 0;
  width: 60px;
  z-index: 2;
}

.c0::before {
  background-color: #FFFFFF;
  bottom: 2px;
  box-shadow: 0 3px 3px 0 rgba(0,20,29,0.2),0 2px 4px 0 rgba(0,20,29,0.15);
  content: "";
  height: 20px;
  position: absolute;
  left: 2px;
  -webkit-transition: -webkit-transform .4s;
  -webkit-transition: transform .4s;
  transition: transform .4s;
  width: 20px;
  z-index: 1;
}

.c0 .c7 {
  position: absolute;
  right: -30px;
  height: 100%;
}

.c3 .c1 {
  color: rgba(0,0,0,0.55);
}

.c4 .c1 {
  color: rgba(0,0,0,0.55);
}

.c4 .c1 {
  color: #FFFFFF;
}

.c5 .c1 {
  color: rgba(0,0,0,0.55);
}

.c5 .c1 {
  color: #FFFFFF;
}

.c6 .c1 {
  color: rgba(0,0,0,0.55);
}

.c6 .c1 {
  color: #FFFFFF;
}

<span
  className="c0"
>
  <div
    className="c1 c2"
    type="off"
  >
    OFF
  </div>
</span>
`;
